app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  validates :username, :email, :password, presence: true
+  validates :username, :email, uniqueness: true
+  validates :username, length: { in: 4..12 }
+  validates :password, length: { in: 6..16 }
+
+  has_many :post
+  has_many :comment
```

Comment:
**Always keep in mind that rails are conventions over configuration**

```ruby
# since we are using the has_many record association, 
# then we must keep in mind that this requires us to make it plural
has_many :posts
has_many :comments

# if we would have to use the has_one record association,
# then we must keep in as a singular
has_one :post
has_one :comment
```